tests/b2share_unit_tests/records/test_record_versions_rest.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of EUDAT B2Share.
+#
+# B2Share is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# B2Share is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with B2Share; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Test B2Share versioning REST API."""
+
+from types import SimpleNamespace
+import json
+import time
+from flask import url_for
+from pprint import pprint
+
+from invenio_search.proxies import current_search
+from b2share.modules.deposit.api import PublicationStates
+
+
+headers = [('Content-Type', 'application/json'),
+           ('Accept', 'application/json')]
+patch_headers = [('Content-Type', 'application/json-patch+json'),
+                 ('Accept', 'application/json')]
+
+
+def test_get_and_search_versions(app, test_records_data, test_users, login_user):
+    """Test search of records with versions."""
+    login = SimpleNamespace()
+    login.normal = lambda client: login_user(test_users['normal'], client)
+    login.owner = lambda c: login_user(test_users['deposits_creator'], c)
+    login.admin = lambda client: login_user(test_users['admin'], client)
+
+    data = test_records_data
+
+    with app.app_context():
+        # create and publish some records in a version chain
+        chain_list = []
+        for _ in range(3):
+            chain = []
+            chain.append(make_record(app, login, data[0]))
+            chain.append(make_record(app, login, data[1], version_of=chain[0]))
+            chain.append(make_record(app, login, data[2], version_of=chain[1]))
+            chain.append(make_record(app, login, data[0], version_of=chain[2]))
+            chain.append(make_record(app, login, data[1], version_of=chain[3]))
+            chain.append(make_record(app, login, data[2], version_of=chain[4]))
+            chain_list.append(chain)
+            print("chain completed")
+
+        time.sleep(1) # wait a bit for elasticsearch
```

Comment:
If we're waiting for the indexing to finish, we could do `current_search_client.indices.flush(*)`

Reply:
👍